src/library/scala/collection/immutable/Map.scala

```diff
@@ -206,11 +206,22 @@ object Map extends MapFactory[Map] {
 
   def empty[K, V]: Map[K, V] = EmptyMap.asInstanceOf[Map[K, V]]
 
-  def from[K, V](it: collection.IterableOnce[(K, V)]): Map[K, V] =
+  def from[K, V](it: IterableOnce[(K, V)]): Map[K, V] =
     it match {
       case it: Iterable[_] if it.isEmpty => empty[K, V]
-      case m: Map[K, V] => m
-      case _ => (newBuilder[K, V] ++= it).result()
+      // Since IterableOnce[(K, V)] launders the variance of K,
+      // identify only our implementations which can be soundly substituted.
+      // For example, the ordering used by sorted maps would fail on widened key type. (scala/bug#12745)
+      //case m: Map[K, V] => m
```

Comment:
Is `//case m: Map[K, V] => m` an example or a missed deletion?

Reply:
It was supposed to be documentary ("this doesn't work"). I will either make it look like a comment or delete it. Thanks.

I also meant to be more specific about what doesn't work for Set.